builtin/providers/azurerm/config.go

```diff
@@ -52,6 +52,7 @@ type ArmClient struct {
 
```

Comment:
It is not necessary to create a new one in case of upgrading. Downgrading is not supported, I don't mind it failing because the error message is quite clear.